parsers/NORDPOOL.py

```diff
@@ -0,0 +1,196 @@
+from datetime import datetime, timedelta
+from enum import Enum
+from functools import lru_cache
+from logging import Logger, getLogger
+
+from requests import Response, Session
+
+from electricitymap.contrib.lib.models.event_lists import PriceList
+from electricitymap.contrib.lib.types import ZoneKey
+
+from .lib.utils import get_token
+
+NORDPOOL_BASE_URL = "https://data-api.nordpoolgroup.com/api/v2/"
+
+# Set this as a global const outside of the functions to cache the token for the current parser instance
+NORDPOOL_TOKEN = None
```

Comment:
I think those two should go together so they are not out of sync. So I would recommend using something like:
```suggestion
@dataclass
class NordpoolToken:
   token: str | None = None
   expiration: timedelta | None = None
TOKEN = NordpoolToken()
```

Reply:
Would this also solve the global issue or would it cause the parsers to fetch a new token on every request?
I would like to avoid querying their auth server more than absolutely necessary.